roles/matrix-jitsi/tasks/validate_config.yml

```diff
@@ -19,6 +19,28 @@
     - "matrix_jitsi_jicofo_auth_password"
     - "matrix_jitsi_jvb_auth_password"
 
+
+- name: Fail if Jitsi internal authentication settings not defined
+  fail:
+    msg: >-
+      You must define the (`{{ item.username }}`) and (`{{ item.password }}`) configuration settings to properly configure an account using Jitsi's internal authentication.
+
+      Please ensure you have defined at least one user and password when using internal authentication.
+      
+      If you're setting up Jitsi for the first time, you may have missed a step.
+      Refer to our setup instructions (docs/configuring-playbook-jitsi.md).
+
+      If you had previously setup Jitsi successfully and are only now facing this error,
+      it means that your installation is most likely using default passwords previously defined by the playbook.
+      These defaults are insecure. Jitsi should be rebuilt with secure values.
+      Refer to the "Rebuilding your Jitsi installation" section in our setup instructions (docs/configuring-playbook-jitsi.md).
+  when:
+    - matrix_jitsi_enable_auth|bool
+    - matrix_jitsi_auth_type == 'internal'
+    - matrix_jitsi_prosody_auth_internal_accounts|length == '0'
+  with_items: "{{ matrix_jitsi_prosody_auth_internal_accounts }}"
```

Comment:
This doesn't make much sense.

You're looping over a the list of accounts, but only `when`: auth is enabled, auth is internal and there are 0 accounts to loop over. I haven't tested this, but I'm guessing it will not do anything.

--------

The result of `matrix_jitsi_prosody_auth_internal_accounts|length` is an integer, so comparing it to a string (`'0'`) is not natural. It will probably get cast and do what you expect anyway, but there's no need to mix types like this.

-------

You seem to be referring to `item`'s username/password values in the `fail` message, saying that one needs to define them.. but that just reveals the passwords and doesn't tell you in which fields to define them.

-------

I'd get rid of the `with_items` here and just have the `when` check, and fix the `|length == '0'` type mismatch thing.

And I'd fix the message to be:

> Enabling internal authentication for Jitsi requires that you define at least one user account in the `matrix_jitsi_prosody_auth_internal_accounts` variable.

----

If you'd like to validate that each entry in the `matrix_jitsi_prosody_auth_internal_accounts` list contains a non-empty username/password, you should probably add a second `fail` task after the first one (the one checking for an empty `matrix_jitsi_prosody_auth_internal_accounts` list), which is like this:

```yaml
- name: Fail if Jitsi internal authentication settings not defined
  fail:
    msg: >-
      Each Jitsi user in `matrix_jitsi_prosody_auth_internal_accounts` needs to have a non-empty `username`/`password`.
  when:
    - matrix_jitsi_enable_auth|bool
    - matrix_jitsi_auth_type == 'internal'
    - item.username == '' or item.password == ''
  with_items: "{{ matrix_jitsi_prosody_auth_internal_accounts }}"
```

I think this is of limited utility though. Who would define a list of users and skip the username or password?

If you want to be even more thorough, you would probably validate that each `item` in this `matrix_jitsi_prosody_auth_internal_accounts` is actually a dictionary/hash, and not something else, guarding against things like this:

```yaml
matrix_jitsi_prosody_auth_internal_accounts: ['username1', 'username2']
```

And being more thorough, you'd wish to guard against it being a variable of a different type even:

```yaml
matrix_jitsi_prosody_auth_internal_accounts: {'username1': 'password', 'username2': 'password'}
```

or

```yaml
matrix_jitsi_prosody_auth_internal_accounts: 4
```

While doing all these checks is nice and may be helpful to someone somewhere, it means we need much more validation code.

Reply:
Thanks for the detailed message @spantaleev :)
You're right that it does what I want, but I didn't go about it the right way. I'll go ahead and work on those changes.
To keep it simple, I'm going to avoid adding extra validation b/c it's certainly not my strong-suit, but will make those corrections.